src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<M> {
```

Comment:
![image](https://user-images.githubusercontent.com/20326858/233832424-c23af87e-00d7-4136-9ff4-06ce7806da3e.png)

Reply:
not fixed, we can still create MachineProducer<String>, lets prevent from it, by adding bounds